doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_rest_handler.md

```diff
@@ -2,126 +2,112 @@
 * [Programmers guide](#section3)
     * [`NGSIRestHandler` class](#section3.1)
 
 ##<a name="section1"></a>Functionality
-###<a name="section1.1"></a>Mapping NGSI events to flume events
-This section explains how a notified NGSI event (an http message) containing context data is converted into a Flume event (an object in memory or a file), suitable for being consumed by any of the Cygnus sinks, thanks to `NGSIRestHandler`.
+###<a name="section1.1"></a>Mapping NGSI events to `NGSIEvent` objects
+This section explains how a notified NGSI event (a http message containing headers and payloa) is used to crate a `NGSIEvent` object, suitable for being consumed by any of the Cygnus sinks, thanks to `NGSIRestHandler`.
 
-It is necessary to remark again this handler is designed for being used by `HttpSource`, the native component of Apache Flume. An http message containing a NGSI-like notification will be received by `HttpSource` and passed to `NGSIRestHandler` in order to create one, and only one, Flume event object to be put in a sink's channel (mainly, these channels are objects in memory, or files).
+It is necessary to remark again this handler is designed for being used by `HttpSource`, the native component of Apache Flume. An http message containing a NGSI-like notification will be received by `HttpSource` and passed to `NGSIRestHandler` in order to create one or more `NGSIEvent` objects (one per notified context element) to be put in a sink's channel (mainly, these channels are objects in memory, but could be files).
 
-On the one hand, the http message containing the NGSI-like notification will be composed of a set of http headers, and a payload. On the other hand, a Flume event object is composed of a set of headers, and a body. As can be seen, there is a quasi-direct translation among http message and Flume event object:
+On the one hand, the http message containing the NGSI-like notification will be composed of a set of http headers, and a payload. On the other hand, the `NGSIEvent` objects are composed of a set of headers as well and an object of type `ContextElement` containing the already parsed version of the context elements within the notification; this parsed version of the notified body is ready for being consumed by other components in the agent architecture, such as interceptors or sinks, thus parsing is just made once.
 
-| http message | Flume event object |
+As can be seen, there is a quasi-direct translation among http messages and `NGSIEvent` objects:
+
+| http message | `NGSIEvent` object |
 |---|---|
-| `Content-Type` header | `content-type` header (discarded from 0.14.0) |
 | `Fiware-Service` header | `fiware-service` header |
 | `Fiware-ServicePath` header | `fiware-servicepath` header |
-| `Fiware-Correlator` header | `fiware-correlator` header |
+| `Fiware-Correlator` header | `fiware-correlator` header. If this header is not sent, the `fiware-correlator` is equals to the `transcation-id` header. |
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1308/commits/7dbe4ab1ef9d2cc343af43c0e0fe9c6a960dc4af